demos/defender/aws_iot_demo_defender.c

```diff
@@ -200,10 +209,11 @@ int RunDefenderDemo( bool awsIotMqttMode,
 {
     int status = EXIT_SUCCESS;
     bool metricsInitStatus = false;
+    bool metricsAccepted = false;
     IotMqttError_t mqttStatus = IOT_MQTT_INIT_FAILED;
     AwsIotDefenderError_t defenderResult = AWS_IOT_DEFENDER_INTERNAL_FAILURE;
     AwsIotDefenderStartInfo_t startInfo = AWS_IOT_DEFENDER_START_INFO_INITIALIZER;
-    const AwsIotDefenderCallback_t callback = { .function = _defenderCallback, .pCallbackContext = NULL };
+    const AwsIotDefenderCallback_t callback = { .function = _defenderCallback, .pCallbackContext = &metricsAccepted };
```

Comment:
Is it thread-safe to pass this locally scoped `metricsAccepted` variable to the callback?

Reply:
The demo has been relying on line 308 to have enough time for the status string to print normally.

Reply:
This would be good to add as a comment to explain the reason for not having any thread-safety logic used for the shared variable

Reply:
Sure